x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -226,18 +226,7 @@ export const shouldTermsAggOnContainer = (groupBy: string | string[] | undefined
 export const flattenAdditionalContext = (
   additionalContext: AdditionalContext | undefined | null
 ): AdditionalContext => {
-  let flattenedContext: AdditionalContext = {};
-  if (additionalContext) {
-    Object.keys(additionalContext).forEach((context: string) => {
-      if (additionalContext[context]) {
-        flattenedContext = {
-          ...flattenedContext,
-          ...flattenObject(additionalContext[context], [context + '.']),
-        };
-      }
-    });
-  }
-  return flattenedContext;
+  return additionalContext ? flattenObject(additionalContext) : {};
```

Comment:
Nice, it looks cleaner now 😍

Shall we also add a test for this util? Something like : https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/utils/flatten_object.test.ts#L11

Reply:
Good idea, I have added tests.